Lib/unittest/case.py

```diff
@@ -603,9 +603,21 @@ def _callSetUp(self):
         self.setUp()
 
     def _callTestMethod(self, method):
-        if method() is not None:
-            warnings.warn(f'It is deprecated to return a value that is not None from a '
-                          f'test case ({method})', DeprecationWarning, stacklevel=3)
+        if (result := method()) is not None:
+            import inspect
+            from unittest.async_case import IsolatedAsyncioTestCase
```

Comment:
This may be nitpicky but you could even defer that import after you checked whether `inspect.iscoroutine` holds or not.